components/layout.js

```diff
@@ -12,23 +12,46 @@ import ListItemText from '@mui/material/ListItemText'
 import MenuIcon from '@mui/icons-material/Menu'
 import PageContainer from '/components/page-container'
 import PropTypes from 'prop-types'
-import { React, useState } from 'react'
+import { React, useState, useMemo } from 'react'
 import SchoolIcon from '@mui/icons-material/School'
 import ThemeProvider from '@mui/material/styles/ThemeProvider'
 import Toolbar from '@mui/material/Toolbar'
 import Typography from '@mui/material/Typography'
 import { createTheme } from '@mui/material/styles'
 import Link from 'next/link'
+import useMediaQuery from '@mui/material/useMediaQuery'
+import { useRouter } from 'next/router'
+import { Fab } from '@mui/material'
+import SearchIcon from '@mui/icons-material/Search'
 
 export default function Layout({ children, window }) {
-  const theme = createTheme({
-    palette: {
-      mode: 'dark',
-    },
-  })
+  const prefersDarkMode = useMediaQuery('(prefers-color-scheme: dark)')
+  const theme = useMemo(
+    () =>
+      createTheme({
+        palette: {
+          mode: prefersDarkMode ? 'dark' : 'light',
+        },
+      }),
+    [prefersDarkMode]
+  )
+  const router = useRouter()
+
   const drawerWidth = 240
+
+  const includeQuery = () => {
+    return router.asPath.includes('courses') ? '' : router.query
+  }
```

Comment:
Note: `schools/[id].js` was replaced with `schools/[schools.js]` so that the query params passed to `/search` work with the search filters.